src/algorithms/models/graph.ts

```diff
@@ -37,36 +41,48 @@ export default class Graph {
     return this.verticesMap.get(id);
   }
 
-  /**
-   * Get function which returns a set of all vertex IDs contained within the graph
-   */
-  getVertexIDs(): Set<number> {
-    const idSet = new Set<number>();
+  removeEdge(from: number, to: number): Edge {
+    const fromVertex = this.getVertex(from);
+    for (const neighbor of fromVertex.neighbors) {
+      if (neighbor.vertex.id === to) {
+        fromVertex.neighbors.delete(neighbor);
+        return neighbor;
+      }
+    }
+  }
+
+  removeVertex(id: number, deletedEdges?: Array<[number, Edge]>): void {
+    // first remove all edges to this vertex
     for (const vertex of this.verticesMap.values()) {
```

Comment:
Fuu, this is gonna be pretty inefficient, right? Isn't there a way to do Yen's without removing a Vertex from the graph? Does this happen often during Yen's?

Reply:
1. Yes, iterating the whole vertices map can turn out to be pretty inefficient, the best thing I can think of would be storing all incoming edges in the vertex itself, which would slow down operations with vertices, but imho not as much as this.
2. Removing vertices from the graph is a feature of Yen's algorithm and there is not much we could do about it. Maybe we could mark the vertices, instead of deleting them, but then you'd have to check every vertex if you can use it... I don't think this would improve the performance.
3. Vertices get removed from the graph when they are part of the root path, then put back and then again for every k.

When we are done with the whole pipeline and have some good results, I'd suggest measuring the time of all the steps in the pipeline and decide what can we do to make it faster. I think we should not focus on this as far as you can get results within Aalborg in <1 minute.